src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/CustomItemEffects.java

```diff
@@ -1144,6 +1369,113 @@ public void getBuildersWandCandidates(
+		int extraMax
+	) {
+		if (target.sideHit != EnumFacing.UP) return;
+		boolean sneaking = Minecraft.getMinecraft().thePlayer.isSneaking();
+		candidatesOld.clear();
+		candidatesOldSorted.clear();
+		LinkedList<BlockPos> candidates = new LinkedList<>();
+		LinkedList<BlockPos> candidatesNew = new LinkedList<>();
+
+		candidatesNew.add(target.getBlockPos());
+
+		double d0 = player.lastTickPosX + (player.posX - player.lastTickPosX) * (double) partialTicks;
+		double d1 = player.lastTickPosY + (player.posY - player.lastTickPosY) * (double) partialTicks;
+		double d2 = player.lastTickPosZ + (player.posZ - player.lastTickPosZ) * (double) partialTicks;
+
+		while (candidatesOld.size() <= MAX_BUILDERS_BLOCKS + extraMax) {
+			if (candidatesNew.isEmpty()) {
+				break;
+			}
+
+			candidates.addAll(candidatesNew);
+			candidatesNew.clear();
+
+			while (!candidates.isEmpty()) {
+				if (candidatesOld.size() > MAX_BUILDERS_BLOCKS + extraMax) break;
+
+				BlockPos candidate = candidates.pop();
+
+				float distSq = (float) ((candidate.getX() + 0.5f - d0) * (candidate.getX() + 0.5f - d0) +
+					(candidate.getY() + 0.5f - d1 - player.getEyeHeight()) *
+						(candidate.getY() + 0.5f - d1 - player.getEyeHeight()) +
+					(candidate.getZ() + 0.5f - d2) * (candidate.getZ() + 0.5f - d2));
+				candidatesOldSorted.computeIfAbsent(distSq, k -> new HashSet<>()).add(candidate);
+
+				candidatesOld.add(candidate);
+
+				float yaw = Minecraft.getMinecraft().thePlayer.getRotationYawHead();
+				Facing facing = Facing.forDirection(yaw);
+				int xOff = facing == Facing.WEST ? -1 : facing == Facing.EAST ? 1 : 0;
+				int zOff = facing == Facing.NORTH ? -1 : facing == Facing.SOUTH ? 1 : 0;
+				if (!sneaking) {
+					if (Minecraft.getMinecraft().theWorld.getBlockState(candidate.add(xOff, 1, zOff)).getBlock() == Blocks.air) {
+
+						BlockPos posNew = candidate.add(xOff, 0, zOff);
+						if (!candidatesOld.contains(posNew) && !candidates.contains(posNew) && !candidatesNew.contains(posNew)) {
+							IBlockState blockNew = Minecraft.getMinecraft().theWorld.getBlockState(posNew.add(0, 1, 0));
+							if (blockNew.getBlock() == Blocks.air) {
+								candidatesNew.add(posNew);
+							}
+						}
+					} else {
+						break;
+					}
+				} else {
+					if (Minecraft.getMinecraft().theWorld.getBlockState(candidate.add(xOff, 0, zOff)).getBlock()
+						== Minecraft.getMinecraft().theWorld.getBlockState(target.getBlockPos()).getBlock()) {
+						BlockPos posNew = candidate.add(xOff, 0, zOff);
+						if (!candidatesOld.contains(posNew) && !candidates.contains(posNew) && !candidatesNew.contains(posNew)) {
+							IBlockState blockNew = Minecraft.getMinecraft().theWorld.getBlockState(posNew.add(0, 0, 0));
+							if (blockNew.getBlock() ==
+								Minecraft.getMinecraft().theWorld.getBlockState(target.getBlockPos()).getBlock()) {
+								candidatesNew.add(posNew);
+							}
+						}
+					} else {
+						break;
+					}
+				}
+
+			}
+		}
+	}
```

Comment:
nopo try writing a single (1(!)) method that is less than 50 lines long challenge (impossble difficulty)